packages/contracts/src/tx-util.ts

```diff
@@ -13,49 +11,23 @@ export class TxUtil {
     account: Account,
     option?: Tx,
   ): Promise<string> {
-    let gas!: number
-    let gasPrice!: string
-    let nonce!: number
-    const promises = [
-      async () => {
-        gas = await tx.estimateGas({
-          ...option,
-          from: account.address,
-        })
-      },
-      async () => {
-        gasPrice = await web3.eth.getGasPrice()
-      },
-      async () => {
-        nonce = await web3.eth.getTransactionCount(account.address, 'pending')
-      },
-    ].map(fetchTask => fetchTask())
-    await Promise.all(promises)
-    const txParams = {
-      nonce: NumString.from(`${nonce}`)
-        .toBytes()
-        .toString(),
-      gasPrice: NumString.from(`${gasPrice}`)
-        .toBytes()
-        .toString(),
-      gasLimit: NumString.from(`${gas}`)
-        .toBytes()
-        .toString(),
+    const [ gas, gasPrice, nonce ] = await Promise.all([
```

Comment:
@wanseob I refactored this when I was debugging something unrelated. Let me know if you're okay with it, using web3 for tx signing is much more clear to me. 🙂

Reply:
This refactoring looks very good to me, I was trying to use `web3.eth.account.ssignTransaction` but didn't notice how to pass the private key into that function, if it does also work well I like this refactoring a lot :+1:  And thanks for prettifying the async things